code/modules/clothing/shoes/clownshoes.dm

```diff
@@ -0,0 +1,207 @@
+/obj/item/clothing/shoes/clown_shoes
+	desc = "The prankster's standard-issue clowning shoes. Damn, they're huge!"
+	name = "clown shoes"
+	icon_state = "clown"
+	inhand_icon_state = "clown_shoes"
+	slowdown = SHOES_SLOWDOWN+1
+	pocket_storage_component_path = /datum/component/storage/concrete/pockets/shoes/clown
+	var/enabled_waddle = TRUE
+	lace_time = 20 SECONDS // how the hell do these laces even work??
+
+/obj/item/clothing/shoes/clown_shoes/Initialize()
+	. = ..()
+	AddComponent(/datum/component/squeak, list('sound/effects/clownstep1.ogg'=1,'sound/effects/clownstep2.ogg'=1), 50)
```

Comment:
Instead of adding the component several times, put a proc here, something like `initialize_squeak()`, and redefine it for each sub-type